kratos/containers/weak_pointer_vector_iterator.h

```diff
@@ -141,7 +141,8 @@ class WeakPointerVectorIterator
 
     typename BaseType::reference dereference() const
     {
-        return *(Kratos::shared_ptr<TDataType>(*(this->base())));
+        typename TDataType::Pointer p( (this->base())->lock() );
```

Comment:
this is the change i am least sure of...

Reply:
seems correct but the creation of p is redundant. 

The old one is using the constructor instead of lock method.